src/app/qgisapp.h

```diff
@@ -212,7 +212,7 @@ class APP_EXPORT QgisApp : public QMainWindow, private Ui::MainWindow
     Q_OBJECT
   public:
     //! Constructor
-    QgisApp( QSplashScreen *splash, bool restorePlugins = true, bool skipBadLayers = false, bool skipVersionCheck = false, const QString &rootProfileLocation = QString(), const QString &activeProfile = QString(), QWidget *parent = nullptr, Qt::WindowFlags fl = Qt::Window );
+    QgisApp( QSplashScreen *splash, bool restorePlugins = true, bool skipBadLayers = false, bool skipVersionCheck = false, const QString &rootProfileLocation = QString(), const QString &activeProfile = QString(), bool enablePython = true, QWidget *parent = nullptr, Qt::WindowFlags fl = Qt::Window );
```

Comment:
This signature is very cumbersome and fragile now (due to implicit pointer conversion without warnings to bool) -- how do you feel about adding an enum flags ``QgisApp::AppOptions`` and replace all these booleans with a single flags argument?

Reply:
Nice idea. Done.